lib/airbrake-ruby.rb

```diff
@@ -11,6 +11,7 @@
 require 'airbrake-ruby/stashable'
 require 'airbrake-ruby/mergeable'
 require 'airbrake-ruby/grouppable'
+require 'airbrake-ruby/keyword_struct'
```

Comment:
While the implementation looks correct to me and it probably allows smooth transition to the new APIs, I'm not a big fan of the approach. We use metaprogramming to mimic the old behaviour but the code looks rather complex. People will need to learn about `KeywordStruct` and I think we could avoid it by refactoring the "model" structs into simple classes. It'll be a little bit more verbose but we wouldn't need to introduce any new concepts/classes. For example:

```diff
diff --git a/lib/airbrake-ruby/queue.rb b/lib/airbrake-ruby/queue.rb
index 275d972..e27b8a6 100644
--- a/lib/airbrake-ruby/queue.rb
+++ b/lib/airbrake-ruby/queue.rb
@@ -5,24 +5,31 @@ module Airbrake
   # @api public
   # @since v4.9.0
   # rubocop:disable Metrics/BlockLength, Metrics/ParameterLists
-  Queue = Struct.new(
-    :queue, :error_count, :groups, :start_time, :end_time, :timing, :time
-  ) do
+  class Queue
     include HashKeyable
     include Ignorable
     include Stashable

+    attr_reader :queue, :error_count, :groups, :start_time, :end_time, :timing,
+                :time
+
     def initialize(
-      queue:,
-      error_count:,
-      groups: {},
-      start_time: Time.now,
-      end_time: start_time + 1,
-      timing: nil,
-      time: Time.now
-    )
+          queue:,
+          error_count:,
+          groups: {},
+          start_time: Time.now,
+          end_time: start_time + 1,
+          timing: nil,
+          time: Time.now
+        )
       @time_utc = TimeTruncate.utc_truncate_minutes(time)
-      super(queue, error_count, groups, start_time, end_time, timing, time)
+      @queue = queue
+      @error_count = error_count
+      @groups = groups
+      @start_time = start_time
+      @end_time = end_time
+      @timing = timing
+      @time = time
     end

     def destination
```

Let me know what you think.

Reply:
That's OK with me. Your call.

I'm going to save KeywordStruct for future reference though in other projects (not here).